operator/pkg/awsprovider/instances/reconciler.go

```diff
@@ -45,29 +45,41 @@ func NewController(ec2api *awsprovider.EC2, autoscaling *awsprovider.AutoScaling
 }
 
 func (c *Controller) Reconcile(ctx context.Context, dataplane *v1alpha1.DataPlane) error {
-	asg, err := c.getAutoScalingGroup(ctx, AutoScalingGroupNameFor(dataplane))
+	subnets, err := c.subnetsFor(ctx, &dataplane.Spec)
 	if err != nil {
-		return fmt.Errorf("getting auto scaling group for %v, %w", dataplane.Spec.ClusterName, err)
+		return fmt.Errorf("getting subnet for %s, %w", dataplane.Spec.ClusterName, err)
+	}
+	return c.CreateResource(ctx, dataplane.Name, &dataplane.Spec, subnets)
+}
+
+func (c *Controller) CreateResource(ctx context.Context, name string, spec *v1alpha1.DataPlaneSpec, subnets []string) error {
```

Comment:
> I think we can avoid all the changes to this file if we just call `Reconcile` and pass `*v1alpha1.DataPlane` from our `EKSTestClusterCreationCommon` controller package we are writing.
> 
> Basically, without making any change just pass Dataplane instead of DataplaneSpec at the runtime and pass subnet selector, it will help us keep all the existing code we have in this file as-is

I think we still need this change. Our use case has a minor difference with the `Reconcile`: we don't want to directly use the `c.subnetsFor`, we want to use `c.subnetsForSelector`.

The  `c.subnetsFor` has fallback on control plane instance subnets, this has a dependency on `kubeClient`, in `nodesWithLabelFor` function call.

I will make a change internally to use the `c.subnetsForSelector` for our new `CreateResource` function instead of using `Reconcile` directly